src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const rules = sourceString.split(';');
+
+  const result = {};
+
+  rules.forEach((rule) => {
+    const [key, value] = rule.split(':').map((part) => part.trim());
+
+    result[key] = value;
+  });
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. In this context, the `forEach` is being used to populate the `result` object, which is fine according to the checklist exception. However, consider using `reduce` to transform the array directly into an object, which aligns with functional programming principles.